### PR TITLE
chore(stdlib)!: Switch to using records for `getInternalStats`

### PR DESCRIPTION
**Pull request by spotandjake**
This cleans up some todos left over from #190 being closed, and switches `getInternalStats` from returning a tuple to a record.

This is marked as breaking as we change the signature of `getInternalStats`

### CODE REVIEW

**Comment by spotandjake** [comment]
Looks like one of the provides snapshots gets updated because of a dependency on the stdlib

**Review by peblair — APPROVED** [review]
Looks good!

File: compiler/test/stdlib/map.test.gr [hunk 1]
```diff
@@ -269,30 +269,29 @@ assert Map.contains(Brick, fa)
 assert !Map.contains(Wood, fa)
 
 // Resizes the map when it grows
-// TODO(#190): Don't use these internals, as they need to change after 190 is fixed
 
 let resize = Map.makeSized(1)
 
 // (nodeCount, bucketLength)
```
Comment (phated) [inline compiler/test/stdlib/map.test.gr:275]: These comments don't make sense anymore

File: compiler/test/stdlib/set.test.gr [hunk 2]
```diff
@@ -235,30 +235,29 @@ assert Array.contains(3, r)
 assert Array.contains(4, r)
 
 // Resizes the map when it grows
-// TODO(#190): Don't use these internals, as they need to change after 190 is fixed
 
 let resize = Set.makeSized(1)
 
 // (nodeCount, bucketLength)
```
Comment (phated) [inline compiler/test/stdlib/set.test.gr:241]: Same as above

File: stdlib/set.gr [hunk 3]
```diff
@@ -24,6 +24,14 @@ abstract record Set<k> {
   mut buckets: Array<Option<Bucket<k>>>,
 }
 
+/**
+ * Represents the internal state of a set.
+ */
+provide record InternalMapStats {
```
Comment (phated) [inline stdlib/set.gr:30]: These are `Set` stats not `Map` stats

File: stdlib/set.gr [hunk 4]
```diff
@@ -452,22 +460,22 @@ provide let intersect = (set1, set2) => {
...
 
-// TODO(#190): Should return a Record type instead of a Tuple
 /**
  * Provides data representing the internal state state of the set.
  *
  * @param set: The set to inspect
  * @returns The internal state of the set
  *
  * @since v0.3.0
+ * @history v0.6.0: switch to returning `InternalMapStats` record
```
Comment (phated) [inline stdlib/set.gr:470]: Same as above

**Review by phated — CHANGES_REQUESTED** [review]
Most notable problem is naming of the record in set

**Comment by spotandjake** [comment]
All Fixed

File: stdlib/set.gr [hunk 5]
```diff
@@ -452,22 +460,22 @@ provide let intersect = (set1, set2) => {
...
 
-// TODO(#190): Should return a Record type instead of a Tuple
 /**
  * Provides data representing the internal state state of the set.
  *
  * @param set: The set to inspect
  * @returns The internal state of the set
  *
  * @since v0.3.0
+ * @history v0.6.0: switch to returning `InternalSetStats` record
```
Comment (phated) [inline stdlib/set.gr:470]:
```suggestion
 * @history v0.6.0: Return `InternalSetStats` record instead of a tuple
```

File: stdlib/map.gr [hunk 6]
```diff
@@ -497,22 +505,22 @@ provide let reject = (predicate, map) => {
...
 
-// TODO(#190): Should return a Record type instead of a Tuple
 /**
  * Provides data representing the internal state state of the map.
  *
  * @param map: The map to inspect
  * @returns The internal state of the map
  *
  * @since v0.2.0
+ * @history v0.6.0: switch to returning `InternalMapStats` record
```
Comment (phated) [inline stdlib/map.gr:515]:
```suggestion
 * @history v0.6.0: Return `InternalMapStats` record instead of a tuple
```

**Review by phated — CHANGES_REQUESTED** [review]
Light rewording on the `@history` attributes

**Review by phated — APPROVED** [review]

**Pull request merged**